src/pip/_internal/models/candidate.py

```diff
@@ -19,17 +18,17 @@ def __init__(self, name, version, link):
 
         super().__init__(
             key=(self.name, self.version, self.link),
-            defining_class=InstallationCandidate
+            defining_class=InstallationCandidate,
         )
 
     def __repr__(self):
         # type: () -> str
         return "<InstallationCandidate({!r}, {!r}, {!r})>".format(
-            self.name, self.version, self.link,
+            self.name, self.version, self.link
```

Comment:
I hate this Black rule so much 😞 Why does it need a trailing comma when arguments are split in separate lines, but actively removes it when all arguments are on one line? This does not make sense.

Reply:
Because black only looks at the CST and doesn't care about whitespace? I don't think there's any way for it to distinguish between the two cases - given that it's meant to not care about existing formatting.

Reply:
FWIW, I agree, but we're at the mercy of the tool here.

Ultimately I think the value provided by auto-formatting out weighs most individual preferences, so I'm fine sticking with what the tool wants.

Would you rather i keep this line as it is now or add the comma back so it splits over multiple lines? I'm fine to go either way with it.